### PR TITLE
Added isSubChannelMember middleware

### PR DESCRIPTION
**Pull request by criticic**
Issue: #58

### CODE REVIEW

File: src/middlewares/isSubChannelMember.middleware.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    try {
+        const userId = req.body.user._id;
+        const subChannelId = parseInt(req.params.id);
+        const subChannel = await SubChannel.findById(subChannelId).populate('channel');
+        if(!subChannel) {
+            return response_404(res, 'SubChannel not found');
+        }
+
+        const channelMember = subChannel.channel.members.find(member => member._id === userId);
+        const workspaceAdmin = subChannel.channel.workspace.createdBy === userId;
```
Comment (automatefreely) [inline src/middlewares/isSubChannelMember.middleware.js:14]: workspace not populated

File: src/middlewares/isSubChannelMember.middleware.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        if(!channelMember && !workspaceAdmin) {
+            return response_403(res, 'You are not authorized to perform this action');
+        }
+
+        req.subChannel = subChannel;
+        req.channel = subChannel.channel;
+        req.workspace = subChannel.channel.workspace;
```
Comment (automatefreely) [inline src/middlewares/isSubChannelMember.middleware.js:21]: same issue

File: src/middlewares/isSubChannelMember.middleware.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const SubChannel = require('../models/subChannel.models.js');
+const {response_403, response_404, response_500} = require('../utils/responseCodes.utils.js');
+
+export async function isSubChannelMember(req, res, next) {
+    try {
+        const userId = req.body.user._id;
+        const subChannelId = parseInt(req.params.id);
```
Comment (automatefreely) [inline src/middlewares/isSubChannelMember.middleware.js:7]: is parseInt required

Comment (criticic) [reply src/middlewares/isSubChannelMember.middleware.js:7]: for params yes

**Pull request merged**